EventFilter/CTPPSRawToDigi/interface/TotemSampicFrame.h

```diff
@@ -144,33 +144,31 @@ class TotemSampicFrame
     inline uint64_t getFPGATimestamp() const {
       uint64_t tmp = 0;
       if ( status_ ) {
-        for ( unsigned short i = 0; i < TotemSampicConstant::fpgaTime_Size; ++i )
-          tmp += totemSampicInfoPtr_[ TotemSampicConstant::fpgaTime_Position + i ] << 8*i;
+        tmp = *( ( const uint64_t* ) ( totemSampicInfoPtr_ + TotemSampicConstant::fpgaTime_Position ) ) & 0xFFFFFFFFFF;
       }
       return tmp;
     }
 
     inline uint16_t getTimestampA() const {
       uint16_t tmp = 0;
       if ( status_ ) {
-        for ( unsigned short i = 0; i < TotemSampicConstant::timestampA_Size; ++i )
-          tmp |= ( totemSampicInfoPtr_[ TotemSampicConstant::timestampA_Position + i ] ) << 8*i;
+        tmp = *( ( const uint16_t* ) ( totemSampicInfoPtr_ + TotemSampicConstant::timestampA_Position ) );
       }
+      tmp = 0xFFF - tmp;
       return grayToBinary<uint16_t> ( tmp );
     }
 
     inline uint16_t getTimestampB() const {
       uint16_t tmp = 0;
       if ( status_ ) {
-        for ( unsigned short i = 0; i < TotemSampicConstant::timestampB_Size; ++i )
```

Comment:
If I am not wrong, all the *_size constants enumerated in TotemSampicConstant now are not used any more. Perhaps you can simplify that ```enum``` by removing all them.